transport/azure-servicebus/src/main/kotlin/AzureServicebusMessageReceiverFactory.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.transport.azureservicebus
+
+import com.azure.identity.DefaultAzureCredentialBuilder
+import com.azure.messaging.servicebus.ServiceBusClientBuilder
+import com.azure.messaging.servicebus.ServiceBusErrorContext
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext
+
+import kotlinx.coroutines.CompletableDeferred
+
+import org.eclipse.apoapsis.ortserver.config.ConfigManager
+import org.eclipse.apoapsis.ortserver.transport.Endpoint
+import org.eclipse.apoapsis.ortserver.transport.EndpointHandler
+import org.eclipse.apoapsis.ortserver.transport.EndpointHandlerResult
+import org.eclipse.apoapsis.ortserver.transport.MessageReceiverFactory
+import org.eclipse.apoapsis.ortserver.transport.json.JsonSerializer
+import org.eclipse.apoapsis.ortserver.utils.logging.runBlocking
+
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+
+private val logger = LoggerFactory.getLogger(AzureServicebusMessageReceiverFactory::class.java)
+
+/**
+ * Implementation of the [MessageReceiverFactory] interface for Azure Servicebus.
+ */
+class AzureServicebusMessageReceiverFactory : MessageReceiverFactory {
+    override val name = "azure-servicebus"
+
+    override suspend fun <T : Any> createReceiver(
+        from: Endpoint<T>,
+        configManager: ConfigManager,
+        handler: EndpointHandler<T>
+    ) {
+        val serializer = JsonSerializer.forClass(from.messageClass)
+        val stopSignal = CompletableDeferred<Unit>()
+
+        fun processMessage(context: ServiceBusReceivedMessageContext) {
+            context.complete() // Acknowledge the message to remove it from the queue.
+            val message = AzureServicebusMessageConverter.toTransportMessage(context.message, serializer)
+
+            MDC.put("traceId", message.header.traceId)
+            MDC.put("ortRunId", message.header.ortRunId.toString())
+
+            if (logger.isDebugEnabled) {
```

Comment:
I'm a little bit confused about these explicit log level checks. When do you actually need them? There are none e.g. for lines 72 and 80.

Reply:
I guess it was there to avoid unnecessary string concatenation if the debug log level is not enabled. I have removed it because the performance impact is irrelevant here and we should finally switch to the same API we use in ORT.